orchestrator/domain/base.py

```diff
@@ -502,6 +503,30 @@ class ProductBlockModel(DomainModel, metaclass=ProductBlockModelMeta):
     owner_subscription_id: UUID
     label: Optional[str] = None
 
+    def dict(
+        self,
+        *,
+        include: Optional[Any] = None,
+        exclude: Optional[Any] = None,
+        by_alias: bool = False,
+        skip_defaults: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> Dict[str, Any]:
+        res = super().dict()
+        parent_ids = []
+        try:
+            for parent in self.parents:
+                if parent and parent.subscription_id != self.owner_subscription_id:
+                    parent_ids.append(parent.subscription_id)
+        except DetachedInstanceError:
```

Comment:
Have you understood why you are getting a DetachtedInstanceError?

Reply:
We try to evaluate a lazy reference to get to it's subscription_id

Reply:
Does it work fine again when you revert to the commit before my suggestion of moving the `== self.owner_subscription_id` into the `@property parent_ids`? Because that seems to have caused this (for reasons I don't yet understand)

Reply:
Nope, I discovered it when I enlarged the scope to testing this branch in the surf orchestrator.